Documentation module

tboro opened the pull request.

Commit:
gitignore, gitkeep update

gitkeep update

gitignore update

Commit:
documentation module

categories - in processing

categories - in processing(2)

categories - in processing(3)

update

categories nested

articles

markup editor for article

markup editor for article

markup rendering in template

markdown editor update

documentation menu

article update

article options update

documentation update

documentation update

article index update

article index update

article index update

article index update

models Article, Category - generate slug beforeSave

categories - renamed js file

articles - sorting, updatePositions

version slug, article version update

update

update

archive for articles for old versions of articles

save to file update

documentation frontend (in processing)

documentation module update

documentation module update

autosave article content changes every 60 sek

documentation breadcrumbs update

documentation searcher

documentation - if article not selected load first

documentation - download as pdf

markup preview update

do not show categories without articles

rockmongo export with documentation collections DEMO

breadcrumbs update

DOMPDF_ENABLE_REMOTE

documentation samples update

save also contentRendered in db

remove parsedown amd markdown-renderer.js, show articles based on contentRendered db field

documentation samples update

update

update

update

update

change PDF generator from dompdf to mpdf

arius86 commented:
Please rebase all commits (https://help.github.com/articles/about-git-rebase/) with proper commit description.

arius86 commented on app/modules/Documentation/controllers/backend/ArticleController.php:
```diff
@@ -0,0 +1,120 @@
+        
+        if (!$article) {
+            $this->throw404($this->_('Page not found.'));
+        }
```
After this check you can just simple put $this->view->article = $article; and remove if ($article) above.

tboro replied:
changed

arius86 commented on app/modules/Documentation/controllers/backend/ArticleController.php:
```diff
@@ -0,0 +1,120 @@
+        $message = 'Failed';
+        $updated = false;
+        
+        if(isset($postData['article'],$postData['content'])) {
```
You can use $this->request->getPost('article','string',fasle) where argument 2 is filter and argument 3 is default value.
There is also $this->request->hasPost('article') to check if variable is present. This comment will also refers to next occurrences of post data manipulation.

tboro replied:
changed

arius86 commented on app/modules/Documentation/controllers/backend/CategoryController.php:
```diff
@@ -0,0 +1,67 @@
+        $status = ($failed==0);
+        if(!$status) $this->response->setStatusCode(500, $message);
+        
+        return $this->jsonResponse(['message' => $message, 'status' => $status]);
```
You don't need 'status' => $status here. The fact that it is response 200 not 500 speaks by itself. Check contact form and how error/success in handled there.

tboro replied:
changed

arius86 commented on app/modules/Documentation/models/Article.php:
```diff
@@ -0,0 +1,71 @@
+    
+    public function makeCopyToArchive($previousState = 1)
+    {
+        if($previousState) $source = Article::findById($this->_id);
```
"if - else" with brackets please.

tboro replied:
changed

arius86 commented:
app/modules/Documentation/mongo-vegas_official_documentation-20150618-143834.gz - this file should be placed in other place than app/modules for sure; maybe in ~/temp or ~/fixtures?

arius86 commented on app/modules/Documentation/models/Article.php:
```diff
@@ -0,0 +1,71 @@
+    public $archival = false;
+    public $parent = null;
+
+    private $archiveFields = ['title','version','category','content', 'contentRendered', 'slug','published','position'];
```
each value in new line please

tboro replied:
changed

arius86 commented on app/modules/Documentation/services/Article.php:
```diff
@@ -0,0 +1,345 @@
+use Documentation\Models\Version as VersionModel;
+use Documentation\Models\Category as CategoryModel;
+
+class Article {    
```
bracket in new line

tboro replied:
changed

arius86 commented on app/modules/Documentation/services/Article.php:
```diff
@@ -0,0 +1,345 @@
+        return false;
+    }
+    
+    public function getArticle($id)
```
Shouldn't be \MongoId $id required here?

tboro replied:
findById accepts string or MongoId. In post and routing params id is saved as string, so why should be here MongoId?
```php 
    /**
         * Find a document by its id (_id)
         *
         * @param string|\MongoId $id
         * @return \Phalcon\Mvc\Collection
         */
        public static function findById($id)
        {
        }
```

arius86 commented on app/modules/Documentation/services/Article.php:
```diff
@@ -0,0 +1,345 @@
+        if(!$version) return null;
+        
+        $model = new ArticleModel();
+        $filter = [];
```
This is $conditions or $query not $filter.

tboro replied:
changed

arius86 commented on app/modules/Documentation/services/Article.php:
```diff
@@ -0,0 +1,345 @@
+            foreach($articles as $categoryId => $articleGroup) {
+                if(!$category || (string) $categoryId == (string) $category->_id) {
+                    if(!empty($articleGroup)) foreach($articleGroup as $article) {
+                        return $article;
```
I don't understand this forech with return. Why it's made like this?

arius86 commented on app/modules/Documentation/services/Article.php:
```diff
@@ -0,0 +1,345 @@
+        
+        return $newArticles;
+    }
+    
```
I think that Categories related methods should be moved to another service.

tboro replied:
done

arius86 commented on app/modules/Documentation/services/Article.php:
```diff
@@ -0,0 +1,345 @@
+        }
+        return $articlesTree;
+    }*/
+
```
Remove commented.

tboro replied:
done

arius86 commented on app/modules/Documentation/views/services/pdf.volt:
```diff
@@ -0,0 +1,11 @@
+<body>
+    TEST PDF
+</body>
+</html>
```
Is this even used?

tboro replied:
removed

arius86 commented:
Please use long syntax (with brackets) for if-else or  $a = $b ? $b : $c; for one line statements. We decided to not use short syntax (without brackets) in vegas-cmf to make code more transparent.

Commit:
documentation module (update based on feedback)

documentation module (update)

Article service refactoring

Article service refactoring

Article service refactoring

refactoring

changes based on feedback from archdevil666pl

changes based on feedback from archdevil666pl (2)

switch version js update

save only supported versions in article record

update

db sample not in repository

arius86 commented:
:+1:

archdevil666pl commented on app/modules/Documentation/controllers/DocumentationController.php:
```diff
@@ -0,0 +1,77 @@
+    public function pdfAction($versionSlug)
+    {
+        $version = $this->serviceManager->getService('documentation:version')->getObjectBySlug($versionSlug);
+        if($version) {
```
What happens otherwise?

tboro replied:
I have changed pdfAction to throw 404 in this case.

archdevil666pl commented on app/modules/Documentation/services/Version.php:
```diff
@@ -0,0 +1,41 @@
+    {
+        $versionsArray = [];
+        $versions = VersionModel::find(['sort' => ['version_id' => 1]]);
+        if(is_array($versions)) foreach($versions as $version) {
```
It's against Vegas code-style.

tboro replied:
changed

archdevil666pl commented on app/modules/Documentation/controllers/backend/ArticleController.php:
```diff
@@ -0,0 +1,130 @@
+    protected $formName = 'Documentation\Forms\Article';
+    protected $modelName = 'Documentation\Models\Article';
+    
+    public function indexAction() {
```
Please add newline before opening braces for class methods.

archdevil666pl commented on app/modules/Documentation/services/Article.php:
```diff
@@ -0,0 +1,234 @@
+
+    private function getServiceVersion()
+    {
+        if($this->serviceVersion == null) {
```
When serviceVersion is 0, this will give true. Please use one of following ways of testing:
- isset()
- ===
- is_null()

archdevil666pl commented on app/modules/Documentation/services/Category.php:
```diff
@@ -0,0 +1,150 @@
+    }
+    
+        
+    public function getChildren($id, $categoriesArray = null)
```
Second param is unused.

Commit:
readme update

Commit:
readme update(2)

arius86 commented on app/modules/Documentation/controllers/DocumentationController.php:
```diff
@@ -0,0 +1,79 @@
+            $filename = 'vegas_'.$versionSlug.'_documentation.pdf';
+            $this->serviceManager->getService('documentation:pdf')->renderPdf($filename,$renderParams);
+        } else {
+            $this->throw404($this->_('Page not found.'));
```
Maybe do:

```
if(!$version) {
    $this->throw404(...)
}
```

And then you don't need else statement.

arius86 commented on app/modules/Documentation/services/Article.php:
```diff
@@ -0,0 +1,236 @@
+        return null;
+    }
+    
+    public function getFirstArticle($versionSlug=null,$categorySlug=null)
```
add spaces in method parameters, we are coding with PSR-2 standards:
https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md
please also check in other methods

arius86 commented on app/modules/Documentation/services/Article.php:
```diff
@@ -0,0 +1,236 @@
+        return $object;
+    }
+    
+    public function getArticleBySlug($versionSlug,$categorySlug,$articleSlug,$onlyPublished=true)
```
add spaces in method parameters, we are coding with PSR-2 standards:
https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md
please also check in other methods

arius86 commented on app/modules/Documentation/services/Category.php:
```diff
@@ -0,0 +1,151 @@
+        return false;
+    }
+    
+    public function getObject($id)
```
For Article service you are using "getArticle" method and here, for same functionality, "getObject". Please unify method names (you can consider `retrieveById($id)`).

arius86 commented on app/modules/Documentation/views/services/versionAsPdf.volt:
```diff
@@ -0,0 +1,51 @@
+        <h3>{{article.title}}</h3>
+        <hr/>
+        <div class="markdown-view">{{article.contentRendered}}</div>
+        {% endfor %}
```
fix indentation

archdevil666pl commented on app/modules/Documentation/controllers/backend/ArticleController.php:
```diff
@@ -0,0 +1,136 @@
+        $article = \Documentation\Models\Article::findById($id);
+        $this->view->article = $article;
+
+        $this->initializeScaffolding();
```
Is this needed when you don't use scaffolding in this action?

archdevil666pl commented on app/modules/Documentation/forms/Article.php:
```diff
@@ -0,0 +1,63 @@
+        $versions = $this->serviceManager->get('documentation:version')->getAll();
+        foreach($versions as $id => $versionId) {
+            $field = new Select('version['.$id.']');
+            $field->setLabel($this->i18n->_('Concerns ver. '.$versionId));
```
Shouldn't we exclude versionID from the translation?

Commit:
no camelCase for Mongo fields

Commit:
documentation module (update based on second feedback)

Commit:
documentation module (update based on second feedback)(2)

Commit:
documentation module (update)(3)

arius86 commented:
:+1: @archdevil666pl ?

archdevil666pl commented:
:+1:
